plugin/gc-app/src/main/java/io/deephaven/app/GcApplication.java

```diff
@@ -0,0 +1,208 @@
+    }
+
+    @Override
+    public ApplicationState create(Listener listener) {
+        final ApplicationState state =
+                new ApplicationState(listener, GcApplication.class.getName(), "Garbage-Collection Application");
+        if (!enabled()) {
+            return state;
+        }
+        final boolean notificationInfoEnabled = notificationInfoEnabled();
+        final boolean poolsEnabled = poolsEnabled();
+        if (!notificationInfoEnabled() && !poolsEnabled()) {
+            return state;
+        }
+        scope = new LivenessScope();
+        try (final SafeCloseable ignored = LivenessScopeStack.open(scope, false)) {
+            if (notificationInfoEnabled) {
+                setNotificationInfo(state);
+            }
+            if (poolsEnabled) {
+                setPools(state);
+            }
+        }
+        install();
+        return state;
+    }
+
+    private void setNotificationInfo(ApplicationState state) {
+        final String name = "notification_info";
+        final SetableStreamPublisher publisher = new SetableStreamPublisher();
+        // noinspection resource
+        final StreamToTableAdapter adapter = new StreamToTableAdapter(GcNotificationConsumer.definition(),
+                publisher, UpdateGraphProcessor.DEFAULT, name);
+        notificationInfoConsumer = new GcNotificationConsumer(publisher.consumer());
+        publisher.setFlushDelegate(notificationInfoConsumer::flush);
+        final Table notificationInfo = adapter.table();
+        state.setField(name, notificationInfo);
+        if (notificationInfoStatsEnabled()) {
+            state.setField(String.format("%s_stats", name), GcNotificationConsumer.stats(notificationInfo));
+        }
+        final int ringSize = notificationInfoRingSize();
+        if (ringSize > 0) {
+            state.setField(String.format("%s_ring", name), RingTableTools.of(notificationInfo, ringSize));
+        }
+    }
+
+    private void setPools(ApplicationState state) {
+        final String name = "pools";
+        final SetableStreamPublisher publisher = new SetableStreamPublisher();
+        // noinspection resource
+        final StreamToTableAdapter adapter = new StreamToTableAdapter(GcPoolsConsumer.definition(), publisher,
+                UpdateGraphProcessor.DEFAULT, name);
+        poolsConsumer = new GcPoolsConsumer(publisher.consumer());
+        publisher.setFlushDelegate(poolsConsumer::flush);
+
+        final Table pools = adapter.table();
+        state.setField(name, pools);
+        if (poolStatsEnabled()) {
+            state.setField(String.format("%s_stats", name), GcPoolsConsumer.stats(pools));
+        }
+    }
+
+    private void install() {
+        for (GarbageCollectorMXBean bean : ManagementFactory.getGarbageCollectorMXBeans()) {
+            if (!(bean instanceof NotificationBroadcaster)) {
+                continue;
+            }
+            ((NotificationBroadcaster) bean).addNotificationListener(this, null, null);
```

Comment:
Maybe you should use a `NotificationFilter`, since you are filtering notifications?